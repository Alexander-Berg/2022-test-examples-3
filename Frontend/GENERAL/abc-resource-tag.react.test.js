import React from 'react';
import { mount } from 'enzyme';

import AbcResourceTag from 'b:abc-resource-tag';

describe('AbcResourceTag', () => {
    it('Should render resource global tag', () => {
        const wrapper = mount(
            <AbcResourceTag
                tag={{ name: { ru: 'tag_name' } }}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('Should render resource service tag', () => {
        const wrapper = mount(
            <AbcResourceTag
                tag={{ name: { ru: 'tag_name' }, service: true }}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });
});
