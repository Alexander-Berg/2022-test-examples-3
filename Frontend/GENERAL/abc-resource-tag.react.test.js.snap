// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbcResourceTag Should render resource global tag 1`] = `
<abc-resource-tag
  tag={
    Object {
      "name": Object {
        "ru": "tag_name",
      },
    }
  }
>
  <div
    className="ta-component abc-resource-tag"
    data-lego="react"
    style={Object {}}
  >
    <abc-resource-tag__text
      key="text"
    >
      <div
        className="ta-component abc-resource-tag__text"
        data-lego="react"
        style={Object {}}
      >
        tag_name
      </div>
    </abc-resource-tag__text>
  </div>
</abc-resource-tag>
`;

exports[`AbcResourceTag Should render resource service tag 1`] = `
<abc-resource-tag
  tag={
    Object {
      "name": Object {
        "ru": "tag_name",
      },
      "service": true,
    }
  }
>
  <div
    className="ta-component abc-resource-tag"
    data-lego="react"
    style={Object {}}
  >
    <icon
      glyph="type-flag"
      key="icon"
      mix={
        Object {
          "block": "abc-resource-tag",
          "elem": "flag",
          "mods": Object {},
        }
      }
    >
      <span
        className="icon icon_glyph_type-flag icon_has-glyph_yes abc-resource-tag__flag"
        data-lego="react"
        style={Object {}}
      >
        <svg
          height="16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 9v5H3V3h10l-2 3 2 3H4z"
            fill="#FF2603"
          />
        </svg>
      </span>
    </icon>
    <abc-resource-tag__text
      key="text"
    >
      <div
        className="ta-component abc-resource-tag__text"
        data-lego="react"
        style={Object {}}
      >
        tag_name
      </div>
    </abc-resource-tag__text>
  </div>
</abc-resource-tag>
`;
